Mage/src/main/java/mage/game/permanent/PermanentImpl.java

```diff
@@ -1932,13 +1932,16 @@ public boolean moveToZone(Zone toZone, Ability source, Game game, boolean flag,
 
     @Override
     public boolean moveToExile(UUID exileId, String name, Ability source, Game game, List<UUID> appliedEffects) {
-        Zone fromZone = game.getState().getZone(objectId);
-        ZoneChangeEvent event = new ZoneChangeEvent(this, source, ownerId, fromZone, Zone.EXILED, appliedEffects);
-        ZoneChangeInfo.Exile zcInfo = new ZoneChangeInfo.Exile(event, exileId, name);
+        if (!game.replaceEvent(GameEvent.getEvent(EventType.EXILE_PERMANENT, objectId, source, controllerId))) {
```

Comment:
There are many places with exile zone moves, must use `GameEvent.EventType.ZONE_CHANGE` to check it instead custom event:
![shot_231028_135045](https://github.com/magefree/mage/assets/8344157/21195408-70a9-4150-96fb-1a372f048630)

Reply:
Done